Interface.C

```diff
@@ -275,6 +280,75 @@ void preciceAdapter::Interface::configureMesh(const fvMesh& mesh)
             }
         }
     }
+
+    else if (locationType_ == LocationType::volumeCenters)
+    {
+        // The volume coupling implementation is based on the coupling with faceCenters
+        // as the volume coupling was tested and developed on a previous development where faceNodes
+        // did not work properly
+        //
+        // The volume coupling implementation considers the mesh points in the volume and
+        // on the boundary patches in order to take the boundary conditions into account
```

Comment:
So, does this identify all points in the volume region + all points on the face centers of the region? Why do we need that?

I would expect that, when coupling a volume, we don't couple the boundaries. Or at least not with the same values. I would expect that the boundaries of the volume are specified by the solver, based on the rest of the domain. This makes mostly sense when coupling only one region of the whole mesh.

Reply:
The coupling includes the cell centers on the whole domain (at this implementation stage) and the face centers for all specified patches - basically adding the cell centers of the volume to the `face nodes`-surface coupling case. 

So in a tutorial case which uses `locations    volumeCenters`, all of the internal fields will be coupled (cell centers), and additionally specified patches can be coupled as well (on face centers).

Reply:
So, if I provide `locations volumeCenters` and don't specify any patches, only the volume region will be modified, without the patches. If I also specify patches, these will also be modified. This sounds reasonable behavior to me, we just need to document it.

Do we need to use different coupling meshes for volumes and patches? I am trying to think of mapping issues, especially now that we have volume mapping as well. @uekerman

Reply:
mmh, not too sure about this (if I understood it correctly). Mixing surface and volume coupling here could confuse more than it helps. There could also be cases where we want to read values in the volume from one solver and additionally read values on the surface from another solver. I would always treat volume and surface as two different meshes in preCICE.
Or is there a specific use case where we would need this feature?

Reply:
So how should we resolve this? Previous applications in that area have used both patches and volume also ([PR#88](https://github.com/precice/openfoam-adapter/pull/88/files) and [PR#97](https://github.com/precice/openfoam-adapter/pull/97/files)), so I suspected it might have been necessary and kept it like that. The channel transport [tutorial](https://github.com/precice/tutorials/pull/315) case I have set up looks very similar whether I couple patches + volume or only volume.